### DISCUSSION TITLE
[Bug]: docker-compose up get  Error: Cannot find module 'firebase/storage'

### DISCUSSION THREAD

**Discussion by SailFlorve** [danny-avila/LibreChat#2949] [Troubleshooting]
### What happened?

LibreChat         | > LibreChat@0.7.2 backend
LibreChat         | > NODE_ENV=production node api/server/index.js
LibreChat         | 
LibreChat         | node:internal/modules/cjs/loader:1148
LibreChat         |   throw err;
LibreChat         |   ^
LibreChat         | 
LibreChat         | Error: Cannot find module 'firebase/storage'
LibreChat         | Require stack:
LibreChat         | - /app/api/server/services/Files/Firebase/crud.js
LibreChat         | - /app/api/server/services/Files/Firebase/index.js
LibreChat         | - /app/api/server/services/Files/strategies.js
LibreChat         | - /app/api/strategies/process.js
LibreChat         | - /app/api/strategies/googleStrategy.js
LibreChat         | - /app/api/strategies/index.js
LibreChat         | - /app/api/server/index.js
LibreChat         |     at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
LibreChat         |     at Module._resolveFilename (/app/node_modules/module-alias/index.js:49:29)
LibreChat         |     at Module._load (node:internal/modules/cjs/loader:986:27)
LibreChat         |     at Module.require (node:internal/modules/cjs/loader:1233:19)
LibreChat         |     at require (node:internal/modules/helpers:179:18)
LibreChat         |     at Object.<anonymous> (/app/api/server/services/Files/Firebase/crud.js:5:71)
LibreChat         |     at Module._compile (node:internal/modules/cjs/loader:1358:14)
LibreChat         |     at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
LibreChat         |     at Module.load (node:internal/modules/cjs/loader:1208:32)
LibreChat         |     at Module._load (node:internal/modules/cjs/loader:1024:12) {
LibreChat         |   code: 'MODULE_NOT_FOUND',
LibreChat         |   requireStack: [
LibreChat         |     '/app/api/server/services/Files/Firebase/crud.js',
LibreChat         |     '/app/api/server/services/Files/Firebase/index.js',
LibreChat         |     '/app/api/server/services/Files/strategies.js',
LibreChat         |     '/app/api/strategies/process.js',
LibreChat         |     '/app/api/strategies/googleStrategy.js',
LibreChat         |     '/app/api/strategies/index.js',
LibreChat         |     '/app/api/server/index.js'
LibreChat         |   ]
LibreChat         | }
LibreChat         | 
LibreChat         | Node.js v20.14.0
LibreChat exited with code 1

### Steps to Reproduce

//

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by SailFlorve** [comment]
LibreChat         | node:internal/modules/cjs/loader:1148
LibreChat         |   throw err;
LibreChat         |   ^
LibreChat         | 
LibreChat         | Error: Cannot find module 'dotenv'
LibreChat         | Require stack:
LibreChat         | - /app/api/server/index.js
LibreChat         |     at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
LibreChat         |     at Module._load (node:internal/modules/cjs/loader:986:27)
LibreChat         |     at Module.require (node:internal/modules/cjs/loader:1233:19)
LibreChat         |     at require (node:internal/modules/helpers:179:18)
LibreChat         |     at Object.<anonymous> (/app/api/server/index.js:1:1)
LibreChat         |     at Module._compile (node:internal/modules/cjs/loader:1358:14)
LibreChat         |     at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
LibreChat         |     at Module.load (node:internal/modules/cjs/loader:1208:32)
LibreChat         |     at Module._load (node:internal/modules/cjs/loader:1024:12)
LibreChat         |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12) {
LibreChat         |   code: 'MODULE_NOT_FOUND',
LibreChat         |   requireStack: [ '/app/api/server/index.js' ]
LibreChat         | }
LibreChat         | 
LibreChat         | Node.js v20.14.0

**Comment by danny-avila** [comment]
What commands did you run? This isn’t happening with the default compose file, using latest images

**Reply by SailFlorve** [reply]
> What commands did you run? This isn’t happening with the default compose file, using latest images

It just docker-compose build --no-cache and docker-compose up,
Here is my docker-compose.override.yml:

version: '3.4'

services:
  api:
    image: librechat
    build:
      context: .
      target: node
  mongodb:
    ports:
      - 27018:27017

**Reply by danny-avila** [reply]
Are you running this within the project directory? When building the image, you need most of the repo files present

**Reply by SailFlorve** [reply]
> Are you running this within the project directory? When building the image, you need most of the repo files present

Of course. I run this in where docker-compose.yml file exists.

**Comment by SailFlorve** [comment]
I  always got sh: cross-env: not found. If I delete cross-env, then I got Error: Cannot find module

**Comment by SailFlorve** [comment]
LibreChat         | > LibreChat@0.7.2 backend
LibreChat         | > cross-env NODE_ENV=production node api/server/index.js
LibreChat         | 
LibreChat         | sh: cross-env: not found
LibreChat exited with code 127

**Comment by danny-avila** [comment]
I can't reproduce this on a clean project directory, here are relevant log outputs and the commands I ran

`git log` output:
```bash
commit 37ae484fbc5f71ea3109296ab2ca88321b0e4acc (HEAD -> main, tag: v0.7.3-rc2, origin/main, origin/HEAD)
Author: Ventz Petkov <901168+ventz@users.noreply.github.com>
Date:   Sat Jun 1 08:51:18 2024 -0400

    🚅 docs: Updated Example for LiteLLM ports and Volume mount (#2941)
    
    * Added necessary "ports" section for it to work by default
    * Added (commented out) example GCP Vertex volume mount for auth config and for ENV variable.
```
`docker-compose.override.yml` file:
```yaml
services:
  api:
    image: librechat
    build:
      context: .
      target: node
```

`docker compose build --no-cache`:

```bash
WARN[0000] The "UID" variable is not set. Defaulting to a blank string. 
WARN[0000] The "GID" variable is not set. Defaulting to a blank string. 
WARN[0000] The "UID" variable is not set. Defaulting to a blank string. 
WARN[0000] The "GID" variable is not set. Defaulting to a blank string. 
WARN[0000] The "UID" variable is not set. Defaulting to a blank string. 
WARN[0000] The "GID" variable is not set. Defaulting to a blank string. 
WARN[0000] /home/danny/LibreChat/docker-compose.yml: `version` is obsolete 
[+] Building 69.8s (12/12) FINISHED                                                                                                                                                                                     docker:default
 => [api internal] load build definition from Dockerfile                                                                                                                                                                          0.0s
 => => transferring dockerfile: 1.17kB                                                                                                                                                                                            0.0s
 => [api internal] load metadata for docker.io/library/node:20-alpine                                                                                                                                                             0.6s
 => [api internal] load .dockerignore                                                                                                                                                                                             0.0s
 => => transferring context: 248B                                                                                                                                                                                                 0.0s
 => CACHED [api 1/7] FROM docker.io/library/node:20-alpine@sha256:928b24aaadbd47c1a7722c563b471195ce54788bf8230ce807e1dd500aec0549                                                                                                0.0s
 => [api internal] load build context                                                                                                                                                                                             2.5s
 => => transferring context: 2.56MB                                                                                                                                                                                               0.5s
 => [api 2/7] RUN apk --no-cache add curl                                                                                                                                                                                         3.5s
 => [api 3/7] RUN mkdir -p /app && chown node:node /app                                                                                                                                                                           0.6s 
 => [api 4/7] WORKDIR /app                                                                                                                                                                                                        0.2s 
 => [api 5/7] COPY --chown=node:node . .                                                                                                                                                                                          8.3s 
 => [api 6/7] RUN     touch .env ;     mkdir -p /app/client/public/images /app/api/logs ;     npm config set fetch-retry-maxtimeout 600000 ;     npm config set fetch-retries 5 ;     npm config set fetch-retry-mintimeout 150  47.0s 
 => [api 7/7] RUN mkdir -p /app/client/public/images /app/api/logs                                                                                                                                                                4.5s 
 => [api] exporting to image                                                                                                                                                                                                      4.8s 
 => => exporting layers                                                                                                                                                                                                           4.7s 
 => => writing image sha256:88f93a6f117513c83e6c894ed1cf68454b823ee4b4a62cc5fbfc6b1f1a3efa86                                                                                                                                      0.0s 
 => => naming to docker.io/library/librechat 
```

Finally, running `docker compose up` with no issues.

**Comment by SailFlorve** [comment]
It may be due to my internet environment, some dependencies may not be able to download properly. I've commented all the code related to Firebase, and the project seems to be running normally for now.
